client/js/nonprofits/donate/parseFields/customAmounts/index.ts

```diff
@@ -0,0 +1,22 @@
+// License: LGPL-3.0-or-later
+import JsonStringParser from './JsonStringParser';
+const { getDefaultAmounts } = require('../../custom_amounts');
+import { splitParam } from '..';
+
+export interface CustomAmount {
+  amount: NonNullable<number>;
+  highlight: NonNullable<string | false>;
+}
+
+export default function parseCustomAmounts(amountsString: string): CustomAmount[] {
+  const defaultAmts = getDefaultAmounts().join();
+
+  if (amountsString.includes('{')) {
+    return new JsonStringParser(`[${amountsString}]`).results;
+  } else {
+    const commaParams = splitParam(amountsString || defaultAmts)
+      .map(Number)
+      .join(',');
+    return new JsonStringParser(`[${commaParams}]`).results;
```

Comment:
I know it's a lot more wordy, but I'd prefer if we handled this similar to the `customFields` and `customField` separation. Sorry :frowning_face:

Reply:
This isn't quite what I intended; I should have been clearer. Look at how `customField` and `customFields` are split up. The old customField parsing is kept in `parseFields/customField/legacy.ts` (or `new.ts`). There are specs just for that to prove that the old parsing style works.

On the other hand, the `parseCustomFields` function does very little: it either sends the fieldsString to JsonStringParser or it sends it to parseSimpleCustomFields which is really the function in `client/js/nonprofits/donate/parseFields/customFields/legacy.ts`. It only handles converting from a comma delimited string to an array and mapping that to the output of `parseCustomField` in `parseFields/customField`. This makes sure that as little as possible happening in every function/class.